packages/block/src/header.ts

```diff
@@ -725,6 +728,10 @@ export class BlockHeader {
    * Returns the hash of the block header.
    */
   hash(): Buffer {
+    if (Object.isFrozen(this)) {
+      return this._cachedHash as Buffer
```

Comment:
I had that originally but typescript was complaining because the type is technically `Buffer | undefined` since it's optional (since it's only used when the object is frozen in this initial implementation). I try to avoid `as` normally, definitely open to other approaches

Reply:
oh i misread your suggestion. that's a good idea, will try it

Reply:
yeah it will gate it as defined with that being the if statement, just a nice simplification :)

Reply:
haha sorry to be picky, but then you actually don't need `Object.isFrozen(this)` then, because cachedHash will only be set if that is true

Reply:
ah yes, good catch

Reply:
updated. thanks for the quick review btw 🙂

Reply:
woot nice looks much better! yeah np late nights :)